### PR TITLE
Sliding overlay

### PR DESCRIPTION
**Pull request by d4vidi**

### CODE REVIEW

File: android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java [hunk 1]
```diff
@@ -81,7 +79,7 @@ private void addScreenStacks() {
 
     private void createAndAddScreens(int position) {
         ScreenParams screenParams = params.tabParams.get(position);
-        ScreenStack newStack = new ScreenStack(activity, getScreenStackParent(), screenParams.getNavigatorId(), this);
+        ScreenStack newStack = new ScreenStack(getActivity(), getScreenStackParent(), screenParams.getNavigatorId(), this);
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java]: Very nice

File: android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java [hunk 2]
```diff
@@ -277,7 +275,7 @@ public void newStack(ScreenParams params) {
         removeView(currentScreenStack.peek());
         currentScreenStack.destroy();
 
-        ScreenStack newStack = new ScreenStack(activity, getScreenStackParent(), params.getNavigatorId(), this);
+        ScreenStack newStack = new ScreenStack(getActivity(), getScreenStackParent(), params.getNavigatorId(), this);
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java]: much performance 😝

File: android/app/src/main/java/com/reactnativenavigation/utils/StubAnimatorListener.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.reactnativenavigation.utils;
+
+import android.animation.Animator;
+
+public class StubAnimatorListener implements Animator.AnimatorListener {
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/utils/StubAnimatorListener.java]: This class is redundant, we can use `AnimatorListenerAdapter`

File: android/app/src/main/java/com/reactnativenavigation/views/slidingOverlay/SlidingOverlay.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            public void onDisplay() {
+                view.setVisibility(View.VISIBLE);
+                animator.animate();
+            }
+        });
+
+    }
+
+    protected ContentView createSlidingOverlayView(SlidingOverlayParams params) {
+        final float heightPixels = ViewUtils.convertDpToPixel(100);
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/views/slidingOverlay/SlidingOverlay.java]: I prefer we let Js dictate the notification height by using `WRAP_CONTENT` and setting a specific height in Jsx. wdyt?

File: android/app/src/main/java/com/reactnativenavigation/animation/PeekingAnimator.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        final int heightPixels = view.getLayoutParams().height;
+
+        // TODO use VisibilityAnimator?
+
+        view.setTranslationY(-heightPixels);
+
+        ObjectAnimator slideIn = ObjectAnimator.ofFloat(view, TRANSLATION_Y, 0);
+        slideIn.setDuration(600);
+        slideIn.setInterpolator(new OvershootInterpolator(0.2f));
+
```
Comment (DanielZlotin) [inline android/app/src/main/java/com/reactnativenavigation/animation/PeekingAnimator.java]: this screams for `slideIn()` method

File: android/app/src/main/java/com/reactnativenavigation/animation/PeekingAnimator.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        ObjectAnimator slideOut = ObjectAnimator.ofFloat(view, TRANSLATION_Y, -heightPixels);
+        slideIn.setDuration(300);
+
```
Comment (DanielZlotin) [inline android/app/src/main/java/com/reactnativenavigation/animation/PeekingAnimator.java]: this too `slideOut()`

File: android/app/src/main/java/com/reactnativenavigation/layouts/BaseLayout.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.reactnativenavigation.layouts;
+
+import android.support.v7.app.AppCompatActivity;
+import android.widget.RelativeLayout;
+
+import com.reactnativenavigation.params.SlidingOverlayParams;
+import com.reactnativenavigation.views.slidingOverlay.SlidingOverlay;
+import com.reactnativenavigation.views.slidingOverlay.SlidingOverlaysQueue;
+
+public abstract class BaseLayout extends RelativeLayout implements Layout {
```
Comment (DanielZlotin) [inline android/app/src/main/java/com/reactnativenavigation/layouts/BaseLayout.java]: we should prefer encapsulation over inheritance. If anything, this should be called `SlidingOverlayLayout` but this is yucky :)

File: android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java [hunk 8]
```diff
@@ -33,9 +33,8 @@
 import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
 import static android.view.ViewGroup.LayoutParams.WRAP_CONTENT;
 
-public class BottomTabsLayout extends RelativeLayout implements Layout, AHBottomNavigation.OnTabSelectedListener {
+public class BottomTabsLayout extends BaseLayout implements AHBottomNavigation.OnTabSelectedListener {
```
Comment (DanielZlotin) [inline android/app/src/main/java/com/reactnativenavigation/layouts/BottomTabsLayout.java]: this means this layout can hold a `slidingOverlayQueue` or something

**Pull request merged**